src/Compilers/CSharp/Portable/FlowAnalysis/PreciseAbstractFlowPass.cs

```diff
@@ -597,15 +597,20 @@ private void AdjustConditionalState(BoundExpression node)
 
         /// <summary>
         /// Visit a general expression, where we will only need to determine if variables are
-        /// assigned (or not). That is, we will not be needing AssignedWhenTrue and
-        /// AssignedWhenFalse.
+        /// assigned (or not).
+        /// If you will be needing WhenTrue and WhenFalse states, set `keepSplit` to `true`.
         /// </summary>
         /// <param name="node"></param>
-        protected BoundNode VisitRvalue(BoundExpression node)
+        protected BoundNode VisitRvalue(BoundExpression node, bool keepSplit = false)
```

Comment:
Consider moving `bool keepSplit` to `NullableWalker` rather than changing the base class. Perhaps change `NullableWalker.VisitRvalueWithResult` instead.